### PR TITLE
Applications-and-Positions-Backend-problems-Fixed: working correctly

### PR DESCRIPTION
**Pull request by ManuelLopezAnido**
Fixing problems in the backend to applications and positions

### CODE REVIEW

**Review by ignacioaristo — APPROVED** [review]
Looks good to me!

File: models/Applications.js [hunk 1]
```diff
@@ -4,9 +4,9 @@ const { Schema } = mongoose;
 
 const applicationSchema = new Schema({
   positionId: { type: mongoose.Schema.Types.ObjectId, ref: 'Positions' },
-  companyId: { type: mongoose.Schema.Types.ObjectId, ref: 'Positions' },
-  postulantsId: [],
-  applicationState: false,
+  companyId: { type: mongoose.Schema.Types.ObjectId, ref: 'Client' },
+  postulantId: { type: mongoose.Schema.Types.ObjectId, ref: 'Postulants' },
+  applicationState: "string",
```
Comment (aquilesmariani) [inline models/Applications.js:9]:
```suggestion
  applicationState: String,
```

File: models/Client.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const mongoose = require('mongoose');
 const { Schema } = mongoose;
 
 const clientSchema = new Schema({
-  companyName: { type: String, required: true },
+  companyName: { type: String, required: false },
```
Comment (aquilesmariani) [inline models/Client.js:6]:
```suggestion
  companyName: String,
```

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Just a few changes to apply

File: models/Positions.js [hunk 3]
```diff
@@ -5,13 +5,13 @@ const { Schema } = mongoose;
...
     jobTitle: { type: String, required: true },
-    clientId: { type: mongoose.Schema.Types.ObjectId, ref: 'Clients' },
+    clientId: { type: mongoose.Schema.Types.ObjectId, ref: 'Client' },
     companyName: { type: String, required: true },
     jobDescription: { type: String, required: true },
     city: { type: String, required: true },
     country: { type: String, required: true },
-    datePosted: { type: Date, required: true },
-    closingDate: { type: Date, required: true },
+    datePosted: { type: String, required: true },
```
Comment (aquilesmariani) [inline models/Positions.js:13]: Did you try parsing the date from the app to the mongoose `Date` type?

**Review by leimeterjoaquin — APPROVED** [review]

**Review by aquilesmariani — APPROVED** [review]
Looks great!

**Pull request merged**